_posts/2025-08-26-strimzi-metrics-reporter.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: post
+title: "Using Strimzi Metrics Reporter to Expose Kafka Metrics in Prometheus Format"
+date: 2025-10-01
+author: Owen Corrigan
+---
+When deploying new Kafka cluster it may be one of your intentions to monitor the cluster using metrics.
+`Apache Kafka®` generates and exposes metrics that reflect how a system is operating. 
+These metrics are useful for monitoring, troubleshooting, tuning, and capacity planning when it comes to running your Kafka cluster.
+In essence, monitoring is crucial to ensure the health and performance of your Kafka clusters.
+`Prometheus` as a metrics monitoring solution has become increasingly common due to its popularity in the cloud-native ecosystem, and it is this format that we use to expose metrics.
+
+The [Strimzi Metrics Reporter](https://github.com/strimzi/metrics-reporter) project implements the Java `MetricsReporter` interface to expose metrics via `Prometheus` and it is available since Strimzi 0.48.0 release.
+
+Up to this point, we have recommended using the `JMX Exporter` as a way to expose Kafka metrics, and we have provided some example yaml files in our [Strimzi Examples folder](https://github.com/strimzi/strimzi-kafka-operator/tree/main/packaging/examples/metrics).
+However, using the JMX Exporter adds unnecessary complexity.
+However, using the JMX Exporter introduces unnecessary complexity. 
+Metrics are first exposed by Kafka through JMX, and then a Java agent scrapes those JMX MBeans, applies mapping rules, and re-exposes them over HTTP for Prometheus. 
+This extra indirection adds several drawbacks:
+* Overhead inside the JVM – The Java agent continuously scrapes and transforms metrics, consuming additional CPU and memory.
+* Unnecessary double handling – Metrics are generated, registered as JMX MBeans, then scraped again, which adds latency and processing steps.
+* Complex and fragile mappings – JMX Exporter’s regex-based rules can rename or restructure metrics. 
+While powerful, this makes it harder to treat metrics like an API, complicating debugging, dashboards, and integration with the Strimzi Operator.
+
+
+The `Strimzi Metrics Reporter` exposes metrics in the Prometheus format directly over HTTP, significantly simplifying the setup and reducing resource consumption.
+It is also a standalone MetricsReporter plugin that can be used with any Kafka Client deployment. 
+
+
+Here, we will discuss Strimzi Metrics Reporter installation and configuration, and guide you on how to use it effectively to monitor your Kafka clusters.
+
+
+## Key Features
```

Comment:
You should use at least one-level lower headings ... so for example `###` here, `####` for the next one, etc. The website template does not translate the headers. So `h1` would be the website title, `h2` is the blog post title, and this would be something like `h3`. So add at least one `#` to all the headers an check the preview. It will look much better.